dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +147,233 @@ async def get_cluster_status(self, req):
                 cluster_status=debug_status(formatted_status_string, error),
             )
 
+    async def get_task_ids_running_in_a_worker(self, worker_id: str) -> List[str]:
+        """
+        Retrieves the task IDs of running tasks associated with a specific worker.
+
+        Args:
+            worker_id (str): The ID of the worker.
+
+        Returns:
+            List[str]: A list containing the task IDs of all the running tasks associated with the worker.
+        """
+
+        option = ListApiOptions(
+            filters=[("worker_id", "=", worker_id), ("state", "=", "RUNNING")],
+            detail=True,
+            timeout=10,
+        )
+        ## Call the state API to get all tasks in a worker
+        tasks_in_a_worker_result = await self._state_api.list_tasks(option=option)
+        tasks_in_a_worker = tasks_in_a_worker_result.result
+
+        ## Get task_id from each task in a worker
+        task_ids_in_a_worker = []
+        if tasks_in_a_worker is not None:
+            task_ids_in_a_worker = [
+                task.get("task_id")
+                for task in tasks_in_a_worker
+                if task and "task_id" in task
+            ]
+        return task_ids_in_a_worker
+
+    async def get_worker_details_for_task(
+        self, task_id: str, attempt_number: int
+    ) -> Tuple[Optional[int], Optional[str]]:
+        """
+        Retrieves worker details for a specific task and attempt number.
+
+        Args:
+            task_id (str): The ID of the task.
+            attempt_number (int): The attempt number of the task.
+
+        Returns:
+            Tuple[Optional[int], Optional[str]]: A tuple containing the worker's PID (process ID),
+            and worker's ID.
+
+        Raises:
+            ValueError: If the task attempt is not running
```

Comment:
This seems unncessary? Or change the function name to maybe something like `get_worker_details_for_running_task`?

Reply:
Good catch, **running** task is better